llama-index-integrations/postprocessor/llama-index-postprocessor-pinecone-native-rerank/README.md

```diff
@@ -0,0 +1,40 @@
+- use rerank models with the pinecone managed vector service to rerank the search results
+- available rerank models from [pinecone](https://app.pinecone.io/organizations/-Nn577_974iRsvC6nVxg/projects/a4fe57a4-b1cc-4a99-bf1d-c35a595cae4a/models)
+
+```python
+import os
+from llama_index.core.schema import NodeWithScore, TextNode
+from llama_index.core.schema import QueryBundle
+from llama_index.postprocessor.pinecone_native_rerank import (
+    PineconeNativeRerank,
+)
+
+os.environ["PINECONE_API_KEY"] = "your-sdk"
+
+txts = [
+    "Apple is a popular fruit known for its sweetness and crisp texture.",
+    "Apple is known for its innovative products like the iPhone.",
+    "Many people enjoy eating apples as a healthy snack.",
+    "Apple Inc. has revolutionized the tech industry with its sleek designs and user-friendly interfaces.",
+    "An apple a day keeps the doctor away, as the saying goes.",
+    "apple has a lot of vitamins",
+]
+
+nodes = [
+    NodeWithScore(node=TextNode(id_=f"vec{i}", text=txt))
+    for i, txt in enumerate(txts)
+]
+
+query_bundle = QueryBundle(
+    query_str="The tech company Apple is known for its innovative products like the iPhone."
+)
+
+reranker = PineconeNativeRerank(top_n=4, model="pinecone-rerank-v0")
```

Comment:
probably should show the api key here, or at least make a note that its set in the env